storage/src/state/ledger.rs

```diff
@@ -1567,6 +1577,14 @@ impl<N: Network> TransactionState<N> {
         Err(anyhow!("Commitment {} is missing in storage", commitment))
     }
 
+    // Returns all the record ciphertexts.
+    fn get_ciphertexts(&self) -> Result<impl Iterator<Item = N::RecordCiphertext> + '_> {
+        Ok(self
+            .commitments
+            .keys()
+            .map(move |commitment| self.get_ciphertext(&commitment).unwrap()))
```

Comment:
why does this function return a `Result` if we're calling `unwrap` on every item? we should probably return an `impl Iterator<Item = Result<N::RecordCiphertext>> + '_` instead

Reply:
I agree with this. The change is already done on the last commit.